framework/http/MemoryStream.php

```diff
@@ -0,0 +1,205 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\http;
+
+use Psr\Http\Message\StreamInterface;
+use yii\base\BaseObject;
+use yii\base\InvalidArgumentException;
+
+/**
+ * MemoryStream uses internal field as a stream source. Thus data associated with this stream exists only in
+ * memory and will be lost once stream is closed.
+ *
+ * Example:
+ *
+ * ```php
+ * $stream = new MemoryStream();
+ *
+ * $stream->write('some content...');
+ * // ...
+ * $stream->rewind();
+ * echo $stream->getContents();
+ * ```
+ *
+ * @author Paul Klimov <klimov.paul@gmail.com>
+ * @since 2.1.0
+ */
+class MemoryStream extends BaseObject implements StreamInterface
```

Comment:
I think current PHP have these for basic objects not extended from anything as well...